modulemd/v2/modulemd-component-rpm.c

```diff
@@ -291,6 +291,80 @@ modulemd_component_rpm_set_property (GObject *object,
 }
 
 
+gboolean
+modulemd_component_rpm_equals (ModulemdComponentRpm *self,
+                               ModulemdComponentRpm *other)
+{
+  g_return_val_if_fail (MODULEMD_IS_COMPONENT_RPM (self), FALSE);
+  g_return_val_if_fail (MODULEMD_IS_COMPONENT_RPM (other), FALSE);
+
+
+  if (self == other)
+    return FALSE;
+
+
+  GStrv arches_as_strv_self = modulemd_component_rpm_get_arches_as_strv (self);
+  GStrv arches_as_strv_other =
+    modulemd_component_rpm_get_arches_as_strv (other);
+
+  // check number of keys
+  if (g_strv_length (arches_as_strv_self) !=
+      g_strv_length (arches_as_strv_other))
+    return FALSE;
+
+
+  // check each element of strv
+  int i = 0;
+  while (i < g_strv_length (arches_as_strv_self))
+    {
+      if (g_strcmp0 (*(arches_as_strv_self + i), *(arches_as_strv_other + i)))
+        return FALSE;
+      i++;
+    }
+
+  GStrv multilib_arches_as_strv_self =
```

Comment:
Another use-case for `modulemd_hash_table_sets_are_equal()`

Reply:
This is giving me a warning because modulemd_hash_table_sets_are_equal takes GHashTable * types as arguments while modulemd_component_rpm_get_multilib_arches_as_strv returns a GStrv type.

Reply:
Same answer as above.